packages/flutter_test/lib/src/window.dart

```diff
@@ -658,21 +666,21 @@ class TestFlutterView implements FlutterView {
   ///   * [resetPhysicalSize] to reset this value specifically
   ///   * [reset] to reset all test values for this view
   @override
-  Size get physicalSize => physicalGeometry.size;
+  Size get physicalSize {
+    // This has to be able to default to `_view.physicalSize` as web_ui handles
+    // `physicalSize` and `physicalGeometry` differently than dart:ui, where
+    // the values are both based off of `physicalGeometry`.
+    return _physicalSize ?? _physicalGeometry?.size ?? _view.physicalSize;
+  }
+  Size? _physicalSize;
   set physicalSize(Size value) {
-    physicalGeometry = physicalGeometry.topLeft & value;
+    _physicalSize = value;
   }
 
-  /// Resets [physicalSize] to the default value for this view.
+  /// Resets [physicalSize] to the default value for this view. This will
+  /// also reset [physicalGeometry] as the values are dependent on one another.
```

Comment:
Nit: the first paragraph should be a one-sentence summary:

```suggestion
  /// Resets [physicalSize] to the default value for this view.
  ///
  /// This will also reset [physicalGeometry] as the values are dependent on one another.
```